### PR TITLE
Display actionables on all proposals

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Show a dot (next to the status) next to actionable proposals in the all proposals list.

# Changes

- new derived store `createIsActionableProposalStore` to get actionable state by proposal id
- render a dot in `ProposalStatusTag` component

# Tests

- remove custom testId `"proposal-status"` from `ProposalStatusTag` because it's outdated
- `createIsActionableProposalStore`
- `ProposalStatusTag` displays actionable
- actionable Nns proposals are marked
- actionable Sns proposals are marked

# Todos

- [ ] Add entry to changelog (if necessary).
Not yet

# Screenshots

## NNS
image

## SNS
image

### CODE REVIEW

File: frontend/src/lib/derived/actionable-proposals.derived.ts [hunk 1]
```diff
@@ -58,3 +62,33 @@ export const actionableProposalSupportedStore: Readable<ActionableProposalSuppor
...
+ * and `undefined` when this information is not available.
+ *
+ * @param proposalId The proposal id to check. Can be undefined for more flexible usage.
+ */
+export const createIsActionableProposalStore = (
+  proposalId: bigint | undefined
+) =>
+  derived(
+    [
+      selectedUniverseIdStore,
```
Comment (dskloetd) [inline frontend/src/lib/derived/actionable-proposals.derived.ts:79]: This makes the behavior of proposal card depend on the selectedUniverse.
Wouldn't that prevent us from making a tab that shows all proposals from all universes?

Have you considered adding a field to `ProposalInfo` indicating if the proposal is actionable?

Comment (mstrasinskis) [reply frontend/src/lib/derived/actionable-proposals.derived.ts:79]: As I see it, this wouldn't prevent us from showing all proposals, because the actionable proposals are a separate lists and we know that all of them are actionable. This is relevant only for the "all proposals" selection.

Comment (dskloetd) [reply frontend/src/lib/derived/actionable-proposals.derived.ts:79]: Would the tab with all proposals not use the `ProposalCard` component?

I think it's a good general principal to all components to be rendered in any context and not depend on the specific page you're on. Is that something you considered? Was it more difficult that way?

I see we also render a `ProposalCard` on the launchpad. Even if that's still not a problem, having this dependency on the page that the component is rendered on, just makes it harder to reason about all the possible consequences.

File: frontend/src/lib/derived/proposals.derived.ts [hunk 2]
```diff
@@ -87,3 +88,26 @@ export const filteredProposals: Readable<ProposalsStore> = derived(
     certified,
   })
 );
+
+export type ActionableProposalInfo = ProposalInfo & {
+  isActionable: boolean | undefined;
+};
+
+export interface FilteredActionableProposalsStore {
```
Comment (mstrasinskis) [inline frontend/src/lib/derived/proposals.derived.ts:96]: @dskloetd, out of curiosity, what do you think about using a tuple for `filteredActionableProposals` (`[ProposalInfo, boolean]`) instead of creating new type `ActionableProposalInfo`?

Comment (dskloetd) [reply frontend/src/lib/derived/proposals.derived.ts:96]: I think it will be slightly less clear what the boolean means.
If you don't want to create a new named type, you can still have names for the individual parts:
```
{ proposalInfo: ProposalInfo; isActionable: boolean}
```

File: frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+export type SnsProposalActionableData = SnsProposalData & {
+  isActionable: boolean | undefined;
+};
+export interface ProjectActionableProposalData {
+  proposals: SnsProposalActionableData[];
+  // certified is an optimistic value - i.e. it represents the last value that has been pushed in store
+  certified: boolean | undefined;
+  // Whether all proposals have been loaded
+  completed: boolean;
```
Comment (dskloetd) [inline frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts:20]: Why is `completed` here? And what's different between SNS and NNS that NNS does not have something similar?

File: frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  [rootCanisterId: string]: ProjectActionableProposalData;
+}
+
+export const snsFilteredActionableProposalsStore = derived<
+  [Readable<SnsProposalsStoreData>, Readable<ActionableSnsProposalsStoreData>],
+  SnsFilteredActionableProposalsStoreData
+>(
+  [snsFilteredProposalsStore, actionableSnsProposalsStore],
+  ([proposalsStore, actionableProposalsStore]) => {
+    return Object.entries(proposalsStore).reduce(
```
Comment (dskloetd) [inline frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts:33]: Would `mapEntries` work here?

File: frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import type { SnsProposalData } from "@dfinity/sns";
+import { derived, type Readable } from "svelte/store";
+
+export type SnsProposalActionableData = SnsProposalData & {
+  isActionable: boolean | undefined;
+};
+export interface ProjectActionableProposalData {
+  proposals: SnsProposalActionableData[];
+  // certified is an optimistic value - i.e. it represents the last value that has been pushed in store
+  certified: boolean | undefined;
```
Comment (dskloetd) [inline frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts:18]: I don't think `certified` is relevant in derived stores, is it?
We usually only use `certified` to decide whether to overwrite the data or whether to fetch new data but this only applies to raw stores. With derived stores we only care about the data itself and just display it.
Is there any code that reads this `certified` field from this derived store?

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts [hunk 6]
```diff
@@ -34,15 +34,15 @@ describe("SnsProposalsList", () => {
   const proposals = [proposal1, proposal2, proposal3];
 
   beforeEach(() => {
-    actionableProposalsSegmentStore.resetForTesting();
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:37]: Why was this removed?

File: frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts [hunk 7]
```diff
@@ -14,10 +15,6 @@ export type SnsProposalActionableData = SnsProposalData & {
 };
 export interface ProjectActionableProposalData {
```
Comment (dskloetd) [inline frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts:16]: Is it useful to have the type `ProjectActionableProposalData` or does it make more sense to use `SnsProposalActionableData[]` directly since `ProjectActionableProposalData` only has 1 field?

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts (hunk 6 above)
Comment (dskloetd) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:37]: Did something stop depending on it because of this PR? Or is this an unrelated drive-by cleanup?

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts [hunk 8]
```diff
@@ -34,15 +34,15 @@ describe("SnsProposalsList", () => {
...
+    page.mock({ data: { universe: mockSnsCanisterIdText } });
   });
 
   it("should render a proposal card per proposal", () => {
     const { queryAllByTestId } = render(SnsProposalsList, {
       props: {
         proposals,
         includeBallots: false,
-        snsName: undefined,
+        snsName: "sns-name",
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:45]: Is the name now required while it wasn't before? Or is this an unrelated cleanup?

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts (hunk 6 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:37]: It's a drive-by cleanup.

File: frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts (hunk 8 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/components/sns-proposals/SnsProposalsList.spec.ts:45]: Totally unrelated drive-by cleanup.

File: frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts (hunk 7 above)
Comment (mstrasinskis) [reply frontend/src/lib/derived/sns/sns-filtered-actionable-proposals.derived.ts]: Not really anymore. Thanks, [updated](https://github.com/dfinity/nns-dapp/pull/4656/files#diff-24cccf2bdd0c2b90e5a90c69c6d85b58fc7eb8ec4222e0d9d0a3429717624992R18).

**Review by dskloetd — APPROVED** [review]
